cloverich/chronicles
# upgrade remark and parse wiki links

- replace remark with (latest versions of) mdast-* utilities
- add test cases / utilities for known issues; address them
- refactor remark-slate-transformer: drop unused syntax, migrate types to `mdast`
- add basic wikilink parsing (via mdast-util-ofm-wikilinks)

In preparation for #132 , where my primary motivation is my Obsidian vault, I need to be able to _parse_ wikilinks. The importer changes I'll add for #132 will handle _resolving them_, which I hope to be straight forward once parsed. I don't plan to support them directly, nor to support any of the more advanced syntax Obsidian offers. Its a bit of overkill already, but its also the only feature I think is needed to support arbitrary markdown directory imports (since I have ~3 sets of slightly different markdown not directories, this will let me unify them into Chronicles)

See Obsidian's [guide on internal links](https://help.obsidian.md/Linking+notes+and+files/Internal+links). Chronicles already supports markdown links ( #124 ), but Obsidian and I think a few others use Wikilinks and I want a _basic_ import path for myself since I plan to use Obsidian for some projects (indefinitely at this point). 

For wikilinks parsing I forked a new Obsidian flavored wikilink parsing lib, because I could not figure out how to import it correctly (esm issues); will try to un-fork once I am fully esm. 

Using the latest mdast-* utilities in general they are now all esm; I think typescripts `esModuleInterop` is doing most of the heavy lifting here, but am not 100% sure. Plan is to continue updaing dependencies then do a final push in #157 

Lots of refactoring of remark-slate-transformer (another fork). Mostly dropping code that is unused, and dropping all types since `mdast` natively exports them. Some related type refactoring. 


This PR will change a lot of the backing markdown aka how notes are persisted. I did a fair amount of testing against my existing notes before / after and think the changes are mostly improvements; anyways I have better test scaffolding to pull in and unit test regressions when I find them. 

Follow-up plan:
- Work out kinks in Wikilnk parsing: Will evaluate as part of #132 
- Further refactor remark-slate-transformer to make it configurable / isolated: #271
- Rest of ESM migration changes #157 
- Unfork ofm-wikilinks #272 (blocked by #157)
- More testing (as issues are discovered)


In general now logging markdown related issues [under the markdown tag](https://github.com/cloverich/chronicles/labels/markdown).
